crates/core/src/vm.rs

```diff
@@ -43,10 +43,27 @@ pub fn build_query<'a>(
         }
     }
 
-    let mut result = get_table(stdb, tx, q)?;
+    let op = if !query.query.is_empty() {
+        Some(query.query.remove(0))
+    } else {
+        None
+    };
+
+    // If the first operation is an index scan, open an index cursor, else a table cursor.
```

Comment:
It's a small thing but I don't love the thrashing the front of the vec `query.query` in the non-index-scan case. Would it be hard to only pop the front in that case, rather than pop and push from the front?